cmd/publicize/main.go

```diff
@@ -0,0 +1,188 @@
+	githubEventServerOptions githubeventserver.Options
+	github                   prowflagutil.GitHubOptions
+	git                      prowflagutil.GitOptions
+
+	webhookSecretFile string
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.BoolVar(&o.dryRun, "dry-run", true, "Dry run for testing. Uses API tokens but does not mutate.")
+
+	fs.StringVar(&o.configPath, "config-path", "/etc/config/config.yaml", "Path to publicize configuration.")
+	fs.StringVar(&o.webhookSecretFile, "hmac-secret-file", "/etc/webhook/hmac", "Path to the file containing the GitHub HMAC secret.")
+
+	fs.StringVar(&o.gitName, "git-name", "", "The name to use on the git commit. Requires --git-email. If not specified, uses the system default.")
+	fs.StringVar(&o.gitEmail, "git-email", "", "The email to use on the git commit. Requires --git-name. If not specified, uses the system default.")
+
+	for _, group := range []flagutil.OptionGroup{&o.github} {
+		group.AddFlags(fs)
+	}
+	o.githubEventServerOptions.Bind(fs)
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Fatalf("cannot parse args: '%s'", os.Args[1:])
+	}
+	return o
+}
+
+func (o *options) Validate() error {
+	if (o.gitEmail == "") != (o.gitName == "") {
+		return errors.New("--git-name and --git-email must be specified")
+	}
+
+	for _, group := range []flagutil.OptionGroup{&o.github} {
+		if err := group.Validate(o.dryRun); err != nil {
+			return err
+		}
+	}
+
+	bytes, err := ioutil.ReadFile(o.configPath)
+	if err != nil {
+		return fmt.Errorf("Couldn't read publicize configuration file: %v", o.configPath)
+	}
+
+	var c Config
+	if err := yaml.Unmarshal(bytes, &c); err != nil {
+		return fmt.Errorf("Couldn't unmarshal publicize configuration: %w", err)
+	}
+
+	if err := c.validate(); err != nil {
+		return err
+	}
+
+	o.bonds = make(map[string]string)
+	for _, bond := range c.Bonds {
+		o.bonds[bond.From] = bond.To
+	}
+
+	if err := o.githubEventServerOptions.DefaultAndValidate(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func main() {
+	o := gatherOptions()
+	if err := o.Validate(); err != nil {
+		logrus.Fatalf("Invalid options: %v", err)
+	}
+
+	secretAgent := &secret.Agent{}
+	if err := secretAgent.Start([]string{o.github.TokenPath, o.webhookSecretFile}); err != nil {
+		logrus.WithError(err).Fatal("Error starting secrets agent.")
+	}
+
+	webhookTokenGenerator := secretAgent.GetTokenGenerator(o.webhookSecretFile)
+	githubTokenGenerator := secretAgent.GetTokenGenerator(o.github.TokenPath)
+	getter := func() sets.String {
+		return sets.NewString([]string{string(webhookTokenGenerator()), string(githubTokenGenerator())}...)
+	}
+	logrus.SetFormatter(logrusutil.NewCensoringFormatter(logrus.StandardLogger().Formatter, getter))
+
+	githubClient, err := o.github.GitHubClient(secretAgent, o.dryRun)
+	if err != nil {
+		logrus.WithError(err).Fatal("Error getting GitHub client.")
+	}
+
+	censor := func(content []byte) []byte {
```

Comment:
Why is this not from the secret agent?

Reply:
Secret agent has `func() []byte` but GitClient wants ` func(content []byte) []byte` /shrug

Reply:
Secret agent has 

```
prow/config/secret/agent.go:func (a *Agent) Censor(content []byte) []byte {
```